core/src/main/kotlin/io/specmatic/conversions/OpenApiSpecification.kt

```diff
@@ -238,7 +239,7 @@ class OpenApiSpecification(
         val updatedScenarios = scenarioInfos.map {
             Scenario(it).copy(
                 dictionary = dictionary.plus(specmaticConfig.parsedDefaultPatternValues()),
-                attributeSelectionPattern = specmaticConfig.attributeSelectionPattern,
+                attributeSelectionPattern = getAttributeSelectionPattern(specmaticConfig),
```

Comment:
The `getAttributeSelectionPattern` method which is from the companion object of SpecmaticConfig should not be used here. Instead, given we need the actual attribute selection options, just return the object in a top-level method.